sansa-ml-spark/src/main/scala/net/sansa_stack/ml/spark/clustering/algorithms/LocalitySensitiveHashing.scala

```diff
@@ -0,0 +1,170 @@
+    val remover = new StopWordsRemover().setInputCol("words").setOutputCol("filtered_words")
+    val removed_df = remover.transform(tokenizedData_Df)
+    return remover.transform(tokenizedData_Df)
+  }
+
+  def vectoriseText(entities: RDD[(String, String)]): DataFrame = {
+    val entityProfile_Df = spark.createDataFrame(entities).toDF("entities", "attributes")
+    val tokenizer = new Tokenizer().setInputCol("attributes").setOutputCol("words")
+    val tokenizedData_Df = tokenizer.transform(entityProfile_Df)
+    val cleanData_Df = removeStopwords(tokenizedData_Df).distinct
+    val cleandfrdd = cleanData_Df.select("filtered_words").distinct.rdd
+    val vocab_size = calculateVocabsize(cleandfrdd)
+    val hashingTf = new HashingTF().setInputCol("filtered_words").
+      setOutputCol("raw_Features").setNumFeatures(Math.round(0.90 * vocab_size).toInt)
+    val isNonZeroVector = udf({ v: Vector => v.numNonzeros > 0 }, DataTypes.BooleanType)
+    val featuredData_hashedDf = hashingTf.transform(cleanData_Df).filter(isNonZeroVector(col("raw_Features")))
+    val idf = new IDF().setInputCol("raw_Features").setOutputCol("features")
+    val idfModel = idf.fit(featuredData_hashedDf)
+    val rescaledHashedData = idfModel.transform(featuredData_hashedDf).
+      filter(isNonZeroVector(col("features")))
+
+    return rescaledHashedData
+  }
+
+  def calculateVocabsize(cleandfrdd: RDD[Row]): Int = {
+    val vocab = cleandfrdd.map(_.mkString).reduce(_ + ", " + _).split(", ").toSet
+    return (vocab.size)
+  }
+
+  def minHashLSH(featuredData_Df: DataFrame): (MinHashLSHModel, DataFrame) = {
+    val mh = new MinHashLSH().setNumHashTables(3).setInputCol("features").setOutputCol("HashedValues")
+    val model = mh.fit(featuredData_Df)
+    val transformedData_Df = model.transform(featuredData_Df)
+    return (model, transformedData_Df)
+  }
+
+  def approxSimilarityJoin(model: MinHashLSHModel, transformedData_Df: DataFrame): Dataset[_] = {
+    val threshold = 0.40
+    return model.approxSimilarityJoin(transformedData_Df, transformedData_Df, threshold)
+  }
+
+  def clusterFormation(dataset: Dataset[_], featuredData_Df: DataFrame) = {
+    val featuredData = featuredData_Df.drop("attributes", "words", "filtered_words")
+
+    val refined_entities_dataset = dataset
+      .filter("datasetA.entities != datasetB.entities")
+      .select(col("datasetA.entities").alias("src"), col("datasetB.entities").alias("dst"))
+
+    import spark.implicits._
+    val c_1 = refined_entities_dataset.select("src")
+    val c_2 = refined_entities_dataset.select("dst")
+    val vertexDF = c_1.union(c_2).distinct().toDF("id")
+
+    val g = GraphFrame(vertexDF, refined_entities_dataset)
+    g.persist()
+    spark.sparkContext.setCheckpointDir(dir_path)
+    
+    // Connected Components are the generated clusters.
+    val connected_components = g.connectedComponents.run()
+
+    //Removing the graphframes checkpoint directory
+    val file_path = Paths.get(dir_path)
```

Comment:
Is this going to work in the cluster? or the dir_path of the checkpoint is always considered to be held on the driver? We should use any file system configurations as soon as it is needed to be distributed across the cluster.

Reply:
why would somebody change the checkpoint dir locally? Also, the whole param isn't documented in the constructor. In my opinion, that should be configured during Spark setup or maybe if you really need it during Spark submit but not in the code.